### DISCUSSION TITLE
Problem with finding antiderviatives

### DISCUSSION THREAD

**Discussion by HegeAaron** [Qalculate/libqalculate#541] [Q&A]
When i try to integrate long expression especially with exponents such as (8x+3)^8+5x^(log48) for instance, it gives me weird long anwsers that are not in the answer options of the MCQs my university publishes. Is there a way to factor it? Pressing always exact doesnt help

**Comment by hanna-kn** [comment]
The problem is that the expression is expanded before integration. The result can be factorized afterwards but this does not result in an expression as simple as if the original where not expanded. Compare with the result of `integrate(8x+3)^y+5x^(log48)`.

**Reply by HegeAaron** [reply]
Thanks! So whats the long term solution?
